src/test/rgw/test_rgw_amqp.cc

```diff
@@ -11,7 +11,12 @@
 
 using namespace rgw;
 
-const std::chrono::milliseconds wait_time(300);
+const std::chrono::milliseconds wait_time(10);
+const std::chrono::milliseconds long_wait_time = wait_time*50;
+auto current_dequeued = 0U;
+// wait for at least one message to be dequeueud, and then wait for all pending answers to be received
+#define WAIT_FOR_MESSAGES do {std::this_thread::sleep_for(wait_time);} while (amqp::get_dequeued() == current_dequeued || amqp::get_inflight() > 0); \
```

Comment:
per the comment, the code should look like
```c++
void wait_until_drained()
{  
  while (amqp::get_dequeued() == 0) {
    std::this_thread::sleep_for(wait_time);
  }
  while (amqp::get_inflight() > 0) {
    std::this_thread::sleep_for(wait_time);
  }
}
```
but the code suggests that once the dequeued message does not change anymore, it will stop waiting.

Reply:
i guess i should fix the comment.
the ``get_dequeued()`` call return a running counter, and in some tests i drain the queue more than once.
so, checking on ``amqp::get_dequeued() == 0`` will return "true" even if no new messages were dequeued (i.e. we didn't even had a chance to send messages out).